search/src/dockit-search.ts

```diff
@@ -0,0 +1,260 @@
+    return this.shadowRoot.querySelector('#search-input');
+  }
+
+  get searchHits(): HTMLUListElement {
+    return this.shadowRoot.querySelector('#search-hits');
+  }
+
+  get searchOverlay(): HTMLDivElement {
+    return this.shadowRoot.querySelector('#search-overlay');
+  }
+
+  static get styles() {
+    return css`
+      #search-box {
+        position: relative;
+        flex-grow: 1;
+        margin-left: 1rem;
+      }
+      #search-box form {
+        z-index: 11;
+        display: flex;
+        align-items: center;
+      }
+      #search-box form label {
+        position: relative;
+        width: 100%;
+      }
+      #search-box form label:before {
+        content: '';
+        z-index: 12;
+        position: absolute;
+        left: 10px;
+        top: 0;
+        bottom: 0;
+        width: 20px;
+        background: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='32' height='32' viewBox='0 0 20 20' fill-rule='evenodd'%3E%3Cpath d='M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z' stroke='gray' fill='none' fill-rule='evenodd' stroke-linecap='round' stroke-linejoin='round' %3E%3C/path%3E%3C/svg%3E")
+          center / contain no-repeat;
+      }
+      #search-overlay {
+        display: none;
+        position: fixed;
+        z-index: 10;
+        background-color: #1f1f1f0f;
+        top: 0;
+        left: 0;
+        right: 0;
+        bottom: 0;
+      }
+      #search-label {
+        padding: 0.5rem;
+        display: flex;
+      }
+      #search-input {
+        border: 1px solid var(--dockit-layout-header-border-color);
+        border-radius: 0.5rem;
+        padding: 0.5rem;
+        padding-left: 2.5rem;
+        width: 100%;
+        z-index: 11;
+        position: relative;
+        background: var(--dockit-layout-bg);
+        font-size: large;
+        color: inherit;
+      }
+      #search-input:focus {
+        outline: 1px solid var(--dockit-layout-accent);
+      }
+      #search-hits {
+        display: none;
+        position: absolute;
+        right: 0;
+        left: 0;
+        list-style: none;
+        overflow: auto;
+        max-height: 70vh;
+        border: 1px solid var(--dockit-layout-header-border-color);
+        background: var(--dockit-layout-bg);
+        z-index: 11;
+        border-radius: 0.5rem;
+        padding: 0.5rem;
+        @media only screen and (max-width: 768px) {
+          position: fixed;
+        }
+      }
+      #search-hits a {
+        overflow: hidden;
+        text-overflow: ellipsis;
+        color: inherit;
+        text-decoration: none;
+        display: block;
+        border-radius: 0.25rem;
+        padding: 0.5rem 0.25rem;
+        border: 1px solid transparent;
+      }
+      #search-hits a:hover {
+        border-color: var(--dockit-layout-accent);
```

Comment:
given you rely on everything from `dockit-layout`, it doesn't seem like this component is usable alone (is it?)
so I'd consider making it part of the `layout` package instead of an individual package
and keep `search` package only for the indexer

Reply:
Honestly I don't mind, niptick detail

Reply:
this is part of the public API of dockit-core, IMO not a nitpick

I'm choosing between
`import '@divriots/dockit-core/search/dockit-search.define.js'` + add in slot manually
or
`import '@divriots/dockit-core/layout/dockit-search.define.js'` + add in slot manually
or even (quite a twist, but given what is happening here, I'd say it's worth considering)
`<dockit-layout activate-search></dockit-layout>` and the rest is automatic

Reply:
Ok I am moving it.